src/reader.rs

```diff
@@ -2115,6 +2115,9 @@ impl ReaderData {
                     }
                     // Else we repaint as normal.
                 }
+                if !self.conf.event.is_empty() {
+                    event::fire_generic(self.parser(), self.conf.event.to_owned(), vec![]);
```

Comment:
what is `fish_prompt` used for today and what do you want to use it for?

Reply:
shell integration https://github.com/kovidgoyal/kitty/issues/7200

Reply:
I think we should first consider implementing `OSC 133` prompt/output marking ourselves, as upstream suggests.
This way terminals and shells don't need to know about each other.
We can probably bake it into the Rust code unless we want to control it from script for whatever reason.
As is, this change is not really appropriate for our only real use of `--on-event fish_prompt`, which is `__fish_enable_bracketed_paste` - though we might need to move that anyway because the enable/disable logic doesn't seem to work in all cases (That's my guess because I made it toggle CSI u too and that sometimes lingers when switching to bash etc.).

Reply:
> OSC 133

That's probably better.